--!strict

export type FaunaConstantsModule = {
	FAUNA_SPAWN_TAG: string,
	FAUNA_ID_ATTRIBUTE: string,
	FAUNA_SPECIES_ATTRIBUTE: string,
	SPAWN_SPECIES_ATTRIBUTE: string,
	SNAKE_ATTACK_STARTED_AT_ATTRIBUTE: string,
	SNAKE_ATTACK_DURATION_ATTRIBUTE: string,
	SNAKE_ATTACK_WINDUP_SECONDS: number,
	SNAKE_ATTACK_TOTAL_SECONDS: number,
	MAX_FAUNA_PER_PLAYER_RADIUS: number,
	PLAYER_FAUNA_RADIUS_STUDS: number,
	AI_TICK_SECONDS: number,
	PLAYER_ATTACK_RANGE_STUDS: number,
	PLAYER_ATTACK_COOLDOWN_SECONDS: number,
	PLAYER_ATTACK_DAMAGE: number,
	FAUNA_SYNC_RADIUS_STUDS: number,
	PATROL_RADIUS_STUDS: number,
	ESCAPE_RADIUS_STUDS: number,
}

local FaunaConstants: FaunaConstantsModule = {
	FAUNA_SPAWN_TAG = "FaunaSpawn",
	FAUNA_ID_ATTRIBUTE = "FaunaId",
	FAUNA_SPECIES_ATTRIBUTE = "SpeciesId",
	SPAWN_SPECIES_ATTRIBUTE = "SpeciesId",
	SNAKE_ATTACK_STARTED_AT_ATTRIBUTE = "SnakeAttackStartedAt",
	SNAKE_ATTACK_DURATION_ATTRIBUTE = "SnakeAttackDuration",
	SNAKE_ATTACK_WINDUP_SECONDS = 0.42,
	SNAKE_ATTACK_TOTAL_SECONDS = 1.0,
	MAX_FAUNA_PER_PLAYER_RADIUS = 8,
	PLAYER_FAUNA_RADIUS_STUDS = 100,
	AI_TICK_SECONDS = 0.25,
	PLAYER_ATTACK_RANGE_STUDS = 10,
	PLAYER_ATTACK_COOLDOWN_SECONDS = 1,
	PLAYER_ATTACK_DAMAGE = 10,
	FAUNA_SYNC_RADIUS_STUDS = 120,
	PATROL_RADIUS_STUDS = 24,
	ESCAPE_RADIUS_STUDS = 55,
}

return FaunaConstants
